AndroidJavaClient/app/src/main/java/com/microsoft/aspnet/signalr/samples/androidjavaclient/MainActivity.java

```diff
@@ -0,0 +1,68 @@
+package com.microsoft.aspnet.signalr.samples.androidjavaclient;
+
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.microsoft.aspnet.signalr.Action;
+import com.microsoft.aspnet.signalr.HubConnection;
+import com.microsoft.aspnet.signalr.HubConnectionBuilder;
+import com.microsoft.aspnet.signalr.LogLevel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        HubConnection hubConnection = new HubConnection("Your URL Here");
```

Comment:
It's because build can throw (in the case when you try to call it more than once). Because of that the code gets really ugly, really fast. I opted to just use the constructor here.

Reply:
We need to either resolve this issue or get comfortable with the "ugly" code :). We should make the `HubConnection` constructor private to match C#. Checked exceptions are somewhat annoying here. Maybe if we make it so `.build()` only throws `Error` rather than `Exception`, we can avoid the extra logic to handle the exception. The `build` method should only throw when the user configures it wrong, so making it unchecked seems reasonable.